assets/src/components/template-inserter/index.js

```diff
@@ -162,8 +169,15 @@ export default compose(
 			getBlocks,
 		} = select( 'core/block-editor' );
 
+		const reusableBlocks = getReusableBlocks();
+
+		const isStoryBlock = ( clientId ) => {
+			const block = getBlock( clientId );
+			return block && 'amp/amp-story-page' === block.name;
+		};
+
 		return {
-			reusableBlocks: getReusableBlocks(),
+			reusableBlocks: reusableBlocks.filter( ( { clientId } ) => isStoryBlock( clientId ) ),
```

Comment:
Since now we're passing through only the filtered blocks already, then perhaps we could name it something else than `reusableBlocks` for clarity?

Reply:
Done it 1b89e313515bb38205e801a7fe1bb8c09af9408f